internal/service/dhcp/filterrelayagent_resource_test.go

```diff
@@ -0,0 +1,783 @@
+					// Test fields with default value
+					resource.TestCheckResourceAttr(resourceName, "is_remote_id", "ANY"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccFilterrelayagentResource_disappears(t *testing.T) {
+	resourceName := "nios_dhcp_filterrelayagent.test"
+	var v dhcp.Filterrelayagent
+	name := "TEST_FILTERRELAYAGENT_RESOURCE_DISAPPEARS"
+	isCircuitID := "NOT_SET"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		CheckDestroy:             testAccCheckFilterrelayagentDestroy(context.Background(), &v),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccFilterrelayagentBasicConfig(name, isCircuitID),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckFilterrelayagentExists(context.Background(), resourceName, &v),
+					testAccCheckFilterrelayagentDisappears(context.Background(), &v),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestAccFilterrelayagentResource_CircuitIdName(t *testing.T) {
+	var resourceName = "nios_dhcp_filterrelayagent.test_circuit_id_name"
+	var v dhcp.Filterrelayagent
+	name := acctest.RandomNameWithPrefix("filterrelayagent")
+	cicuitIdName := "CIRCUIT_ID_NAME_TEST"
+	isCircuitId := "MATCHES_VALUE"
+	updateCircuitIdName := "CIRCUIT_ID_NAME_TEST_UPDATED"
+	isCircuitIdSubstring := "false"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccFilterrelayagentCircuitIdName(name, cicuitIdName, isCircuitId, isCircuitIdSubstring),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckFilterrelayagentExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "circuit_id_name", cicuitIdName),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccFilterrelayagentCircuitIdName(name, updateCircuitIdName, isCircuitId, isCircuitIdSubstring),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckFilterrelayagentExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "circuit_id_name", updateCircuitIdName),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccFilterrelayagentResource_CircuitIdSubstringLength(t *testing.T) {
+	var resourceName = "nios_dhcp_filterrelayagent.test_circuit_id_substring_length"
+	var v dhcp.Filterrelayagent
+	name := acctest.RandomNameWithPrefix("filterrelayagent")
+	isCircuitID := "MATCHES_VALUE"
+	isCircuitIdSubstring := "true"
+	circuitIdSubstringOffset := "0"
+	circuitIdName := "CIRCUIT_ID_NAME_TEST"
+	isCircuitIdSubstringLength := fmt.Sprintf("%d", len(circuitIdName))
```

Comment:
```suggestion
	circuitIdSubstringLength := fmt.Sprintf("%d", len(circuitIdName))
```
Also Please rename updateIsCircuitIdSubstringLength to updateCircuitIdSubstringLength
as these are int variables